### ISSUE TITLE
Private or public API for Push

### ISSUE THREAD

**Issue by tcitworld** [nextcloud/notifications#1055]
Needed for https://github.com/nextcloud/server/issues/22379 (no need for notification content, only push). Also a leftover from https://github.com/nextcloud/notifications/pull/59, so it might have other uses.

* Move logic from `PushController` to a new `PushService`, allowing to register through
* Refactor talk specifics (also needed for https://github.com/nextcloud/notifications/issues/594)
* Allow to send custom payloads?

**Comment by nickvergessen** [comment]
> "Allow to send custom payloads"

I don't think this would help with anything. The push proxy can only push to mobile apps of the same developer account. 


> Refactor talk specifics (also needed for #594)

It's a one liner, the line is even mentioned in the ticket
